### PR TITLE
ONT-551 add export account

### PR DESCRIPTION
**Pull request by AlverLyu**
Export accounts to another wallet file. Support re-encryption for lower performance devices.

### CODE REVIEW

File: account/file_store.go [hunk 1]
```diff
@@ -22,11 +22,15 @@ import (
...
 	"encoding/json"
+	"errors"
+	"fmt"
+
 	"github.com/ontio/ontology-crypto/keypair"
 	//"github.com/ontio/ontology/core/types"
-	"github.com/ontio/ontology/common"
 	"io/ioutil"
 	"os"
+
```
Comment (Honglei-Cong) [inline account/file_store.go]: go format imports ??

File: account/file_store.go [hunk 2]
```diff
@@ -151,5 +172,48 @@ func (this *WalletData) Load(path string) error {
...
+func (this *WalletData) ToDefaultSecurity(passwords [][]byte) error {
+	return this.reencrypt(passwords, nil)
+}
+
+func (this *WalletData) reencrypt(passwords [][]byte, param *keypair.ScryptParam) error {
+	if len(passwords) != len(this.Accounts) {
+		return errors.New("not enough passwords for the accounts")
+	}
+	keys := make([]*keypair.ProtectedKey, len(this.Accounts))
+	for i, v := range this.Accounts {
```
Comment (Honglei-Cong) [inline account/file_store.go]: it seems the following two loops can merged together?

File: account/file_store.go [hunk 3]
```diff
@@ -91,6 +95,23 @@ func NewWalletData() *WalletData {
 	}
 }
 
+func (this *WalletData) Dump() *WalletData {
```
Comment (Honglei-Cong) [inline account/file_store.go:96]: question: is this Copy() ?

File: cmd/account_cmd.go [hunk 4]
```diff
@@ -424,3 +435,43 @@ func accountImport(ctx *cli.Context) error {
...
+				pwd, err := password.GetPassword()
+				if err != nil {
+					fmt.Println(err)
+				} else {
+					passwords[i] = pwd
+					break
+				}
+			}
+		}
+		wallet = wallet.Dump()
```
Comment (Honglei-Cong) [inline cmd/account_cmd.go]: this copied walletData to new instance, should client.WalletData be updated?

File: cmd/account_cmd.go [hunk 5]
```diff
@@ -424,3 +435,43 @@ func accountImport(ctx *cli.Context) error {
...
+		err := wallet.ToLowSecurity(passwords)
+		if err != nil {
+			return fmt.Errorf("export failed: %s", err)
+		}
+	}
```
Comment (Honglei-Cong) [inline cmd/account_cmd.go]: if not to low security, to default security? or to original security?

File: account/client.go [hunk 6]
```diff
@@ -313,7 +315,9 @@ func (this *ClientImpl) getAccount(accData *AccountData, passwd []byte) (*Accoun
 	if !accData.VerifyPassword(passwd) {
 		return nil, fmt.Errorf("verify password failed")
 	}
+	keypair.SetScryptParam(this.walletData.Scrypt)
 	privateKey, err := keypair.DecryptPrivateKey(&accData.ProtectedKey, passwd)
+	keypair.SetScryptParam(nil)
```
Comment (Honglei-Cong) [inline account/client.go:320]: question: changed to default security params here?

File: account/file_store.go (hunk 3 above)
Comment (AlverLyu) [reply account/file_store.go:96]: Yes, deep copy.

File: cmd/account_cmd.go (hunk 5 above)
Comment (AlverLyu) [reply cmd/account_cmd.go]: Same as the original wallet.

File: account/client.go (hunk 6 above)
Comment (AlverLyu) [reply account/client.go:320]: Every encryption/decryption operation set the parameters as needed, and reset to default afterwards.

File: account/file_store.go (hunk 3 above)
Comment (laizy) [reply account/file_store.go:96]: Unfortunately，it is not deep copy.  value returned by v.GetKeyPair will share internal data with the original one. and `Dump` is usually used to print some debug message.  `Clone` or `Copy` is a better one?

Comment (laizy) [reply account/file_store.go:96]: if you do need deep copy , you can use  `json.Marshal` and `json.UnMarshal`

Comment (AlverLyu) [reply account/file_store.go:96]: It does not matter actually. This function is used with reencrypt.

**Pull request closed without merging**